File: leetcode/shortest-path-to-get-all-keys/solution_test.py
import sys, os
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
sys.path.insert(0, os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", ".."))
from typing import List
from tools.leetcode import test, summary
from solution import Solution

sol = Solution()
test(sol.shortestPathAllKeys(["@a"]), 1, "One key, adjacent")
test(sol.shortestPathAllKeys(["@.a..", "###.#", "b.A.B"]), 8, "Example 1")

summary()

File: leetcode/shortest-path-to-get-all-keys/solution_test.cpp
#include "../../tools/leetcode.h"
#include "solution.cpp"

int main() {
    {
        vector<string> g = {"@a"};
        test(Solution{}.shortestPathAllKeys(g) == 1, "One key, adjacent");
    }
    {
        vector<string> g = {"@.a..", "###.#", "b.A.B"};
        test(Solution{}.shortestPathAllKeys(g) == 8, "Example 1");
    }
    return summary();
}
